driver-core/src/main/com/mongodb/internal/connection/AwsAuthenticator.java

```diff
@@ -193,7 +193,7 @@ public void dispose() {
         @NonNull
         String getUserName() {
             String userName = credential.getUserName();
-            if (userName == null) {
+            if (userName == null && (userName = System.getenv("AWS_ACCESS_KEY_ID")) == null) {
```

Comment:
Nit: use nested if statements, as you did for the two changes below

Reply:
Done